From c84977a867bd43435b1ab327f44a48bf7c8dc09d Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 30 Apr 2015 17:33:58 +0800
Subject: [PATCH 376/462] atomisp2: Clean up unused code

Change-Id: I1f39acfdf5e1cfa26c061de96eef9b4ff71ee04d
Fixes-Issue: #JIMINAN-34893
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/53730
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Jian Xu Zheng <jian.xu.zheng@intel.com>
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_cmd.c        |  2 +-
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c     |  2 --
 .../media/pci/atomisp2/atomisp_driver/atomisp_subdev.c     | 14 --------------
 .../media/pci/atomisp2/atomisp_driver/atomisp_subdev.h     |  3 ---
 4 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index c08a368..4063c86 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -6156,7 +6156,7 @@ int atomisp_get_pipe_id(struct atomisp_video_pipe *pipe)
 		else
 			return CSS_PIPE_ID_PREVIEW;
 	} else if (pipe == &asd->video_out_capture) {
-		if (asd->copy_mode && !asd->copy_mode_format_conv)
+		if (asd->copy_mode)
 			return IA_CSS_PIPE_ID_COPY;
 		else
 			return CSS_PIPE_ID_CAPTURE;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 126c3d7..8080fb9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2466,8 +2466,6 @@ static enum ia_css_pipe_mode __pipe_id_to_pipe_mode(
 		if (mipi_info && mipi_info->input_format ==
 			ATOMISP_INPUT_FORMAT_YUV420_8_LEGACY)
 			return IA_CSS_PIPE_MODE_YUVPP;
-		if (asd->copy_mode_format_conv)
-			return IA_CSS_PIPE_MODE_CAPTURE;
 		return IA_CSS_PIPE_MODE_COPY;
 	case IA_CSS_PIPE_ID_PREVIEW:
 		return IA_CSS_PIPE_MODE_PREVIEW;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index 0f988e3..731144c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -132,20 +132,6 @@ bool atomisp_subdev_format_conversion(struct atomisp_sub_device *asd,
 		&& !atomisp_is_mbuscode_raw(src->code);
 }
 
-bool atomisp_subdev_copy_format_conversion(struct atomisp_sub_device *asd,
-				      unsigned int source_pad)
-{
-	struct v4l2_mbus_framefmt *sink, *src;
-
-	sink = atomisp_subdev_get_ffmt(&asd->subdev, NULL,
-				       V4L2_SUBDEV_FORMAT_ACTIVE,
-				       ATOMISP_SUBDEV_PAD_SINK);
-	src = atomisp_subdev_get_ffmt(&asd->subdev, NULL,
-				      V4L2_SUBDEV_FORMAT_ACTIVE, source_pad);
-
-	return sink->code != src->code;
-}
-
 uint16_t atomisp_subdev_source_pad(struct video_device *vdev)
 {
 	return vdev->entity.links[0].source->index;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
index ae62cdc..d37acef 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
@@ -380,7 +380,6 @@ struct atomisp_sub_device {
 	unsigned int mipi_frame_size;
 
 	bool copy_mode; /* CSI2+ use copy mode */
-	bool copy_mode_format_conv; /* CSI2+ copy with format conversion */
 	bool yuvpp_mode;	/* CSI2+ yuvpp pipe */
 
 	int raw_buffer_bitmap[ATOMISP_MAX_EXP_ID/32 + 1]; /* Record each Raw Buffer lock status */
@@ -413,8 +412,6 @@ const struct atomisp_in_fmt_conv *atomisp_find_in_fmt_conv_compressed(
 bool atomisp_subdev_format_conversion(struct atomisp_sub_device *asd,
 				      unsigned int source_pad);
 uint16_t atomisp_subdev_source_pad(struct video_device *vdev);
-bool atomisp_subdev_copy_format_conversion(struct atomisp_sub_device *asd,
-					   unsigned int source_pad);
 
 /* Get pointer to appropriate format */
 struct v4l2_mbus_framefmt
-- 
1.9.1

